src/esp/metadata/MetadataMediator.cpp

```diff
@@ -99,45 +105,54 @@ bool MetadataMediator::createSceneDataset(const std::string& sceneDatasetName,
   if (exists) {
     // check if not overwrite and exists already
     if (!overwrite) {
-      ESP_WARNING() << "Scene Dataset" << sceneDatasetName
-                    << "already exists.  To reload and overwrite existing "
-                       "data, set overwrite to true. Aborting.";
-      return false;
+      ESP_WARNING(Mn::Debug::Flag::NoSpace)
+          << "Scene Dataset `" << sceneDatasetName
+          << "` already exists. To reload and overwrite existing "
+             "data, set overwrite to true. Using existing Scene Dataset.";
+      // Treat as if created, since it exists.
+      return true;
     }
     // overwrite specified, make sure not locked
     sceneDatasetAttributesManager_->setLock(sceneDatasetName, false);
   }
-  // by here dataset either does not exist or exists but is unlocked.
+  // by here dataset either does not exist or exists but is unlocked and
+  // overwrite is specified. attempt to create new/overwrite
   auto datasetAttribs =
       sceneDatasetAttributesManager_->createObject(sceneDatasetName, true);
-  if (datasetAttribs == nullptr) {
-    // not created, do not set name
-    ESP_WARNING() << "Unknown dataset" << sceneDatasetName
-                  << "does not exist and is not able to be created.  Aborting.";
-    return false;
-  }
+  // Failure here means some catastrophic error attempting to create the dataset
+  // attributes. Should fail code
+  ESP_CHECK(datasetAttribs,
+            "Scene Dataset `"
+                << Mn::Debug::nospace << sceneDatasetName << Mn::Debug::nospace
+                << "` does not exist and is unable to be "
+                   "created using that name. Verify the given file name.");
+
   // if not null then successfully created
-  ESP_DEBUG() << "Dataset" << sceneDatasetName << "successfully created.";
   // lock dataset to prevent accidental deletion
+  // should only be removable via MetadataMediator::removeSceneDataset.
   sceneDatasetAttributesManager_->setLock(sceneDatasetName, true);
+  ESP_DEBUG(Mn::Debug::Flag::NoSpace) << "Dataset `" << sceneDatasetName
+                                      << "` successfully created and locked.";
+
```

Comment:
Perhaps some more of these logs could be very verbose, as they trace normal function behavior.